common/client.go

```diff
@@ -318,15 +325,27 @@ func (rc *RestClient) Get(url string, result interface{}) error {
 // GetServiceConfig retrieves configuration for a given service from the root service.
 func (rc *RestClient) GetServiceConfig(rootServiceUrl string, svc Service) (*ServiceConfig, error) {
 	rootIndexResponse := &RootIndexResponse{}
+
 	err := rc.Get(rootServiceUrl, rootIndexResponse)
 	if err != nil {
 		return nil, err
 	}
+
+	if rc.config.Credential.Type != CredentialNone {
```

Comment:
I believe that the need to get the token can be entirely driven by the server response. This would also allow us to later add a different token system, maybe via pre-installed certificates, which do not require anything from auth at all.